### PR TITLE
Clean up hooks code

### PR DESCRIPTION
**Pull request by esune**
Signed-off-by: Emiliano Suñé <emiliano.sune@gmail.com>

### CODE REVIEW

File: api/src/utils/hooks.ts [hunk 1]
```diff
@@ -52,33 +48,19 @@ export async function verifyJWT(context: HookContext) {
     return Promise.reject(new Forbidden("The authorization header is missing"));
   }
   const token = authHeader.split(" ")[1];
```
Comment (amanji) [inline api/src/utils/hooks.ts:50]: Should the extract function be used here too?

File: api/src/utils/hooks.ts [hunk 2]
```diff
@@ -52,33 +48,19 @@ export async function verifyJWT(context: HookContext) {
...
-  try {
-    decoded = verify(token, keys[0].getPublicKey(), {
-      algorithms: context.app.get("authentication").algorithms,
-    });
-  } catch (error) {
-    throw new NotAuthenticated(`Authentication failed: ${error.message}`);
-  }
+  const keys = await getAuthSigningKeys(context);
+  decodeIdToken(token, keys, context);
   return context;
```
Comment (amanji) [inline api/src/utils/hooks.ts:53]: Does returning the context pass the hook? Should an error be instead thrown if authentication fails?

File: api/src/utils/hooks.ts [hunk 3]
```diff
@@ -52,33 +48,19 @@ export async function verifyJWT(context: HookContext) {
...
-    if (!authHeader) {
+    const token = extractIdToken(
+      context.params.headers?.authorization as string
+    );
+    if (!token) {
       throw new Forbidden("The authorization header is missing");
     }
-    const token = authHeader.split(" ")[1];
-
     const decoded = decode(token) as {
```
Comment (amanji) [inline api/src/utils/hooks.ts:64]: What happens to the block of code following this if `decoded === undefined`

File: api/src/utils/hooks.ts (hunk 1 above)
Comment (esune) [reply api/src/utils/hooks.ts:50]: Good catch, I missed that one.

File: api/src/utils/hooks.ts (hunk 2 above)
Comment (esune) [reply api/src/utils/hooks.ts:53]: Yes, returning the context would pass the hook. An exception would be raised in the `verifyIdToken` function if something fails during verification and break the chain.

File: api/src/utils/hooks.ts (hunk 3 above)
Comment (esune) [reply api/src/utils/hooks.ts:64]: This should raise an exception that would break the chain and fail the request.

**Review by amanji — APPROVED** [review]

**Pull request merged**